Sources/Bond/Collections/ObservableArray.swift

```diff
@@ -295,30 +296,30 @@ extension MutableObservableArray {
 }
 
 extension MutableObservableArray where Item: Equatable {
-  
+
   public func replace(with array: [Item], performDiff: Bool) {
-    if performDiff {
-      lock.lock()
+    lock.lock(); defer { lock.unlock() }
 
+    if performDiff {
       let diff = self.array.extendedDiff(array)
+      let patch = diff.patch(from: self.array, to: array)
```

Comment:
@tonyarnold are you 100% about this? There is a difference in what `extendedDiff` and `patch` produce. If I remember correctly, when doing table/collection **batch** updates, one should use diff, and when **not** doing batch updates one should use patch. I might be wrong.

Wokalsi also appears to be using diff for batch updated: https://github.com/wokalski/Diff.swift/blob/master/Sources/Diff%2BUIKit.swift